credoai/prism/comparators/metric_comparator.py

```diff
@@ -0,0 +1,106 @@
+"""Comparators for metric type containers"""
+from typing import Callable, List, Literal, Optional
+
+import numpy as np
+from pandas import DataFrame, concat
+
+from credoai.evaluators.utils.validation import check_instance
+from credoai.evidence import MetricContainer
+from credoai.prism.comparators.comparator import Comparator
+from credoai.utils.common import ValidationError
+
+
+class MetricComparator(Comparator):
+    """
+    Class for comparing metric evidence objects
+
+    Supported comparisons for Metrics include difference, maximal values for each metric,
+    and minimal values for each metric. Comparisons are evaluated on intersection of the
+    metrics represented in the two provided MetricContainer objects. Comparison values for
+    non-intersecting metrics will be NoneType wrapped in output container.
+
+    Inputs:
+        EvidenceContainers: dictionary of {name_of_model: MetricContainer} key-value pairs
+
+    Output, stored in a LensComparison object, is result of comparisons.
+    """
+
+    OPERATIONS = {
+        "diff": lambda x, ref: x - ref,
+        "ratio": lambda x, ref: x / ref,
+        "perc": lambda x, ref: x * 100 / ref,
+        "perc_diff": lambda x, ref: ((x - ref) / ref) * 100,
+    }
+
+    def __init__(
+        self,
+        EvidenceContainers: List[MetricContainer],
+        ref: Optional[str] = None,
```

Comment:
What happens if there is no `ref`? Do we want to default to just the first evidencecontainer?